tests/light/functional_tests/filterx/test_filterx.py

```diff
@@ -73,6 +73,76 @@ def create_config(config, filterx_expr, msg="foobar"):
     return (file_true, file_false)
 
 
+def test_otel_logrecord_int32_setter_getter(config, syslog_ng):
+    (file_true, file_false) = create_config(
+        config, """
+                                            $olr = otel_logrecord_new();
+                                            $olr.dropped_attributes_count = ${values.int};
+                                            $MSG = $olr.dropped_attributes_count; """,
+    )
+    syslog_ng.start(config)
+
+    assert file_true.get_stats()["processed"] == 1
+    assert "processed" not in file_false.get_stats()
+    assert file_true.read_log() == "5\n"
+
+
+def test_otel_logrecord_string_setter_getter(config, syslog_ng):
+    (file_true, file_false) = create_config(
+        config, """
+                                            $olr = otel_logrecord_new();
+                                            $olr.severity_text = "${values.str}";
+                                            $MSG = $olr.severity_text; """,
+    )
+    syslog_ng.start(config)
+
+    assert file_true.get_stats()["processed"] == 1
+    assert "processed" not in file_false.get_stats()
+    assert file_true.read_log() == "string\n"
+
+
+def test_otel_logrecord_bytes_setter_getter(config, syslog_ng):
+    (file_true, file_false) = create_config(
+        config, """
+                                            $olr = otel_logrecord_new();
+                                            $olr.trace_id = ${values.bytes};
+                                            $MSG = $olr.trace_id; """,
+    )
+    syslog_ng.start(config)
+
+    assert file_true.get_stats()["processed"] == 1
+    assert "processed" not in file_false.get_stats()
+    assert file_true.read_log() == "binary whatever\n"
+
+
+def test_otel_logrecord_datetime_setter_getter(config, syslog_ng):
+    (file_true, file_false) = create_config(
+        config, """
+                                            $olr = otel_logrecord_new();
+                                            $olr.time_unix_nano = ${values.datetime};
+                                            $MSG = $olr.time_unix_nano; """,
+    )
+    syslog_ng.start(config)
+
+    assert file_true.get_stats()["processed"] == 1
+    assert "processed" not in file_false.get_stats()
+    assert file_true.read_log() == "1701353998.123000+00:00\n"
+
+
+def test_otel_logrecord_body_setter_getter(config, syslog_ng):
+    (file_true, file_false) = create_config(
+        config, """
+                                            $olr = otel_logrecord_new();
+                                            $olr.body = "my body";
+                                            $MSG = $olr.body; """,
+    )
+    syslog_ng.start(config)
+
+    assert "processed" not in file_true.get_stats()
+    assert file_false.get_stats()["processed"] == 1
+    assert file_false.read_log() == "foobar\n"
```

Comment:
Why is this `foobar` instead of `my body`?

Reply:
I think I understand. This does not work because `body` is an `AnyValue`, which is not handled right now.

We should not write a test for this, or should put a comment that it is not the expected final behavior, and must be changed when `AnyValue` is supported.

Reply:
this test check the failure at the moment due to "not implemented" error. it's not a real test, more likely a placeholder for the body test and some experiment around the light tests framework. i can remove it if you want guys.